compiler/ras/Tree.cpp

```diff
@@ -2865,6 +2865,7 @@ int32_t childTypes[] =
    TR::Address,                    // TR::checkcast
    TR::Address,                    // TR::checkcastAndNULLCHK
    TR::Address,                    // TR::New
+   TR::NoType,                    // TR::newvalue
```

Comment:
This is surprising - the opcode declares itself to return TR::Address as does the opcode design - why is this TR::NoType?

Reply:
According to the name and comment, `childTypes` holds the types of the _children_ of a node with a given opcode, not the type of its result. `TR::NoType` is the same as the entry for `TR::acall`, and seems to indicate that the type is not statically known.

Now that we know that the symref for `newvalue` has to be an allocation helper, and the first child will have to pick out the type, it might be better to specify that the first child is address-typed like `TR::acalli` does with `TR::NoType | (TR::Address<<8)`.

Reply:
@Charlmzz FYI ^^